styx-common/src/test/java/com/spotify/styx/util/GrpcContextUtilTest.java

```diff
@@ -0,0 +1,166 @@
+    final CompletableFuture<String> value = new CompletableFuture<>();
+    final Runnable runnable = () -> value.complete(TEST_KEY.get());
+    Context.current().withValue(TEST_KEY, "foobar")
+        .run(() -> sut.submit(runnable, "quux"));
+    assertThat(value.get(30, SECONDS), is("foobar"));
+  }
+
+  @Test
+  public void submitCallableShouldPropagateContext() throws Exception {
+    final Callable<String> callable = TEST_KEY::get;
+    final Future<String> value = Context.current().withValue(TEST_KEY, "foobar")
+        .call(() -> sut.submit(callable));
+    assertThat(value.get(30, SECONDS), is("foobar"));
+  }
+
+  @Test
+  public void invokeAllShouldPropagateContext() throws Exception {
+    final Callable<String> callable = TEST_KEY::get;
+    final List<Future<String>> futures = Context.current().withValue(TEST_KEY, "foobar")
+        .call(() -> sut.invokeAll(ImmutableList.of(callable)));
+    assertThat(futures.get(0).get(30, SECONDS), is("foobar"));
+  }
+
+  @Test
+  public void invokeAllWithTimeoutShouldPropagateContext() throws Exception {
+    final Callable<String> callable = TEST_KEY::get;
+    final List<Future<String>> futures = Context.current().withValue(TEST_KEY, "foobar")
+        .call(() -> sut.invokeAll(ImmutableList.of(callable), 30, SECONDS));
+    assertThat(futures.get(0).get(30, SECONDS), is("foobar"));
+  }
+
+  @Test
+  public void invokeAnyShouldPropagateContext() throws Exception {
+    final Callable<String> callable = TEST_KEY::get;
+    final String result = Context.current().withValue(TEST_KEY, "foobar")
+        .call(() -> sut.invokeAny(ImmutableList.of(callable)));
+    assertThat(result, is("foobar"));
+  }
+
+  @Test
+  public void invokeAnyWithTimeoutShouldPropagateContext() throws Exception {
+    final Callable<String> callable = TEST_KEY::get;
+    final String result = Context.current().withValue(TEST_KEY, "foobar")
+        .call(() -> sut.invokeAny(ImmutableList.of(callable), 30, SECONDS));
+    assertThat(result, is("foobar"));
+  }
+
+  @Test
+  public void shouldPropagateContextForCfRunAsync() throws Exception {
+    final CompletableFuture<String> value = new CompletableFuture<>();
+    final Runnable runnable = () -> value.complete(TEST_KEY.get());
+    Context.current().withValue(TEST_KEY, "foobar")
+        .run(() -> CompletableFuture.runAsync(runnable, sut));
+    assertThat(value.get(30, SECONDS), is("foobar"));
+  }
+
+  @Test
+  public void shouldPropagateContextForCfSupplyAsync() throws Exception {
+    final Supplier<String> get = TEST_KEY::get;
+    final CompletableFuture<String> cf = Context.current().withValue(TEST_KEY, "foobar")
+        .call(() -> CompletableFuture.supplyAsync(get, sut));
+    assertThat(cf.get(30, SECONDS), is("foobar"));
+  }
+
+  @Test
+  public void shouldNotBeConstructable() throws ReflectiveOperationException {
+    final Constructor<GrpcContextUtil> constructor = GrpcContextUtil.class.getDeclaredConstructor();
```

Comment:
Aiming for 100% coverage. :)

Reply:
Yeah. We should have some reusable utility `verifyNotInstantiable(Class<?> cls)` method